Code/Framework/AzCore/Tests/Math/RayTests.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AZTestShared/Math/MathTestHelpers.h>
+#include <AzCore/Math/LineSegment.h>
+#include <AzCore/Math/Ray.h>
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/UnitTest/TestTypes.h>
+
+namespace UnitTest
+{
+    TEST(MATH_Ray, Ray_CreateFromOriginAndDirection)
+    {
+        AZ::Ray ray(AZ::Vector3(0.0f, 0.5f, 12.0f), AZ::Vector3(3.0f, 0.0f, 4.0f).GetNormalized());
+        EXPECT_THAT(ray.GetOrigin(), IsClose(AZ::Vector3(0.0f, 0.5f, 12.0f)));
+        EXPECT_THAT(ray.GetDirection(), IsClose(AZ::Vector3(3.0f, 0.0f, 4.0f).GetNormalized()));
+    }
+
+    TEST(MATH_Ray, Ray_CreateFromLineSegment)
+    {
+        AZ::LineSegment segment(AZ::Vector3(0.0f, 0.5f, 12.0f), AZ::Vector3(15.0f, 0.0f, 3.0f));
+        AZ::Ray ray = AZ::Ray::CreateFromLineSegment(segment);
+
+        EXPECT_THAT(ray.GetOrigin(), IsClose(AZ::Vector3(0.0f, 0.5f, 12.0f)));
+        EXPECT_THAT(ray.GetDirection(), IsClose((AZ::Vector3(15.0f, 0.0f, 3.0f) - AZ::Vector3(0.0f, 0.5f, 12.0f)).GetNormalized()));
+    }
+
+    TEST(MATH_Ray, Ray_CreateDirectionNonNormalizedFail)
+    {
+        AZ_TEST_START_TRACE_SUPPRESSION;
+        AZ::Ray rayUnused = AZ::Ray(AZ::Vector3(), AZ::Vector3(3, 0, 0));
+        AZ_TEST_STOP_TRACE_SUPPRESSION(1);
```

Comment:
The [AZ_MATH_ASSERT](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Math/MathUtils.h#L30-L37) only occurs in Debug configuration.

When the test are run in profile configuration, the AZ::Ray constructor math assert does not occur.
Therefore the test fails because the expected amount of asserts of 1 did not occur.

Here are two ways to solve the issue.

1. Surround the `TEST(MATH_Ray, Ray_CreateDirectionNonNormalizedFail)` test in an `#if defined(AZ_MATH_ASSERT)` block
2. Use the [AZ_TEST_STOP_TRACE_SUPPRESSION_NO_COUNT](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/UnitTest/UnitTest.h#L307) macro to allow for 0 or more assert messages to occur

Reply:
AZ_MATH_ASSERT is always defined so I don't think that will work